src/client_side.cc

```diff
@@ -1540,8 +1544,6 @@ bool ConnStateData::serveDelayedError(Http::Stream *context)
                 clientReplyContext *repContext = dynamic_cast<clientReplyContext *>(node->data.getRaw());
                 assert (repContext);
 
-                request->hier = sslServerBump->request->hier;
-
                 // Create an error object and fill it
```

Comment:
TODO:

- [ ] Mention that this PR is reversing 2012 commit 06b4e9c8 logic (reimplemented in 2017 commit d8165775).
- [ ] Consider keeping `%I` and `%W` expansions intact by adding a new ALE field their code can use to report the same information without copying HierarchyLogEntry (or by reusing some existing field?)
- [ ] Assess possible impact on ACLServerCertificateStrategy::match(): Is it possible to construct an example where the lack of `checklist->al->hier.tcpServer` would alter how Squid behaves in affected test cases? For example, if access_log ACLs restrict logging to transactions with a particular TLS server, will the problematic GET be dropped from access log after this fix?